n2survey/plot/simple_comparison.py

```diff
@@ -0,0 +1,241 @@
+        # 'add_questions' that are also compared with the compare_with_question
+        # in 'plot'()-function of survey.py
+        totalbar = None
+        if answer_suppress:
+            for answer in answer_suppress:
+                if answer not in percentage_correlated_answers:
+                    print(f"{answer} does not exist for this question")
+                else:
+                    percentage_correlated_answers.pop(answer)
+        percentages_for_comparison.append(percentage_correlated_answers.copy())
+
+    x = []
+    y = []
+    for percentage_correlated_answers in percentages_for_comparison:
+        for entry in percentage_correlated_answers:
+            x.append(entry)
+            y.append(percentage_correlated_answers[entry])
+    return x, y
+
+
+def form_bar_positions(df, bar_positions=False, totalbar=None, answer_suppress=False):
+    """
+    forms a complete list of bar positions for all bars, also the not
+    specified ones.
+    You can specify bar positions in a list, but if you
+    want to specify the last bar you have to specify every bar before that,
+    Maybe this can be improved for more user friendlieness.
+    --> maybe exchange to a bar distance dictionary?
+    if dictionary to complex for standard user, maybe list of tuples?
+    [('Total',1.5),('Yes', 0.2),] etc.
+    with (nameofbar, distance_from_previous_bar)
+    At the moment we calculate PERCENTAGE a second time in this function,
+    but the slowing of the whole plot function is neglectable.
+    """
+    # start list of bar_positions
+    bar_positions_complete = bar_positions or [0]
+    count = 0
+    number_of_answers = 0
+    for answer_combinations in df:
+        percentage_correlated_answers = get_percentages(
+            answer_combinations, totalbar=totalbar
+        )
+        for answer in answer_suppress:
+            if answer not in percentage_correlated_answers:
+                print(f"{answer} does not exist for this question")
+            else:
+                percentage_correlated_answers.pop(answer)
+        number_of_answers = number_of_answers + len(percentage_correlated_answers)
+        totalbar = None
+        if count > 0 and len(bar_positions_complete) == number_of_answers - len(
+            percentage_correlated_answers
+        ):
+            bar_positions_complete.append(max(bar_positions_complete) + 1.5)
+        while len(bar_positions_complete) < number_of_answers:
+            bar_positions_complete.append(max(bar_positions_complete) + 1)
+        count = count + 1
+    return bar_positions_complete
+
+
+def filter_answer_sequence(x, answer_sequence):
+    """
+    removes answers from sequence that where supressed while creating x
+    """
+    all_answer_sequence = []
+    for answer_list in answer_sequence:
+        all_answer_list = answer_list.copy()
+        for answer in all_answer_sequence.copy():
+            if answer not in x:
+                all_answer_list.remove(answer)
```

Comment:
There seems to be something weird with the way answer suppression is implemented. If I run 
```python
s.plot(
        "A7", 
        compare_with="A4", 
        add_questions=["A5", "A6"],
        answer_suppress=["No Answer", "Man"],
    )
```
the "Man" column is removed from the plot, as expected
![image](https://user-images.githubusercontent.com/77344869/155748920-5fc20100-c617-4a71-b12f-62d1038a8a47.png)

However, if I swap the first and the third questions (A6 and A7), such that I run 
```python
s.plot(
        "A6", 
        compare_with="A4", 
        add_questions=["A5", "A7"],
        answer_suppress=["No Answer", "Man"],
    )
```
I get an exception saying `ValueError: list.remove(x): x not in list`. Is it never intended to remove columns from the first question?

Reply:
Yes, this was a bug, which it seems I fixed, now it works.